### PR TITLE
Added a small part of BTree implementation

### PR DESCRIPTION
**Pull request by devenbansod**
@rajaths589 : could you please review this ? (nothing much to review, but still)

Signed-off-by: Deven Bansod devenbansod.bits@gmail.com

### CODE REVIEW

File: src/btree.h [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+		);
+
+		root = NULL;
+
+		// now ready for operation
+	}
+
+	~BTree() {
+		// save the header to the file on disk
+		buffered_file_internal->writeHeaderPersistent();
```
Comment (rajathshashidhara) [inline src/btree.h]: what is writeHeaderPersistent() ?

File: include/buffer.h [hunk 2]
```diff
@@ -61,6 +61,7 @@ class BufferedFile
 	void writeHeader(const void* header_buffer);
 	long allotBlock();
 	void deleteBlock(long block_number);
+	int getFileDescriptor();
```
Comment (rajathshashidhara) [inline include/buffer.h]: why is this needed ?

File: src/btree.h (hunk 1 above)
Comment (devenbansod) [inline src/btree.h]: if we have made any changes to the header in memory i.e. `buffered_file_internal->header`, before deleting it from the memory, we write the header to the file and then delete it.

File: include/buffer.h (hunk 2 above)
Comment (devenbansod) [inline include/buffer.h]: Actually we don't need it. 
Sorry. Will remove it.

**Comment by devenbansod** [comment]
Don't merge it just as of now. I will let you know once it's ready.

File: src/btree.h (hunk 1 above)
Comment (rajathshashidhara) [inline src/btree.h]: not required. when BufferedFile destructor is called, this will automatically happen

Comment (devenbansod) [inline src/btree.h]: okay. great! Will remove it then!

**Comment by devenbansod** [comment]
Looks okay to merge @rajaths589 ?

**Comment by rajathshashidhara** [comment]
you can maybe squash the commits ?

**Comment by devenbansod** [comment]
Yes. I am just rebasing it over the latest master and then will squash and merge.

**Comment by devenbansod** [comment]
Merging!
